README.md
# Custom Sign-In Buttons :3
## A package designed to create sign in buttons in an easier and more customizable way.

> ## Inspired in [flutter_signin_buttons](https://pub.dev/packages/flutter_signin_button)


## Installation

Add to pubspec.yaml.

```yaml
dependencies:
  ...
  custom_signin_buttons: ^1.0.0
```

## Usage Example

import custom_signin_buttons.dart

```dart
import 'package:custom_signin_buttons/custom_signin_buttons.dart';
```

### For built-in buttons.

```dart
SignInButton(
  Buttons.Google,
)

// with custom text
SignInButton(
  button: Button.Discord,
  text: 'Inicia con Discord',
),
```

### For mini buttons.

```dart
SignInButton(
  Buttons.Facebook,
  mini: true,
)
```
### For small buttons.

```dart
SignInButton(
  Buttons.Facebook,
  small: true,
)
```

### For self-build buttons.

```dart
CustomSignInButton(
  text: 'Sign In With Email',
  customIcon: Icons.email,
  buttonColor: Colors.blue,
  iconColor: Colors.grey,
  textColor: Colors.grey,
  mini: false,
)

CustomSignInButton(
  customIcon: FontAwesomeIcons.envelopeOpen,
  buttonColor: Colors.blue,
  iconColor: Colors.grey,
  mini: true,
),
```

``` onPressed(){} ``` isn't required

### Built-in Buttons contain

```dart
enum Button {
  Apple,
  AppleDark,
  Discord,
  Email,
  Facebook,
  FacebookNew,
  GitHub,
  Google,
  GoogleBlue,
  GoogleBlack,
  GooglePlay,
  Instagram,
  InstagramGradient,
  LinkedIn,
  Pinterest,
  PlayStation,
  PlayStationGradient,
  Reddit,
  RedditGradient,
  Steam,
  SteamGradient,
  Spotify,
  SpotifyLight,
  TikTok,
  Twitch,
  TwitchLight,
  Twitter,
  Xbox,
  XboxDark,
  Youtube,
  YoutubeDark,
}
```


<img src="https://github.com/DarkHachi/custom_signin_buttons/raw/master/showcase.PNG" width="300">


CHANGELOG.md
# 1.0.2

- Change ```useText``` to ```showText```.
- Removed ```textColor```, ```iconColor``` and ```buttonColor``` for built-in buttons (those properties were not used).
- Removed ```spanish``` for built-in buttons (unnecessary property).
- Readme.md updated.

# 1.0.1

- Fix Asset Images.

# 1.0.0

- Initial release.


test/custom_signin_buttons_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:custom_signin_buttons/custom_signin_buttons.dart';

void main() {}


example/lib/main.dart
// ignore_for_file: use_key_in_widget_ructors, prefer__ructors, avoid_unnecessary_containers,prefer__literals_to_create_immutables,prefer_final_fields,, use_key_in_widget_constructors
import 'package:flutter/material.dart';
import 'package:custom_signin_buttons/custom_signin_buttons.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        debugShowCheckedModeBanner: false,
        title: 'Material App',
        home: ButtonPage());
  }
}

class ButtonPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final size = MediaQuery.of(context).size;
    return Scaffold(
      body: Container(
        margin: EdgeInsets.symmetric(horizontal: 30),
        child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              SignInButton(
                button: Button.AppleDark,
                borderRadius: 15,
                //// buttonColor: Colors.green,
                //// iconColor: Colors.green,
                height: 40,
                iconSize: 30,
                mini: false,
                onPressed: () {},
                splashColor: Colors.pink,
                small: false,
                // textColor: Colors.blue,
                showText: true,
                textSize: 15,
                width: 300,
              ),
              SignInButton(
                button: Button.AppleDark,
                small: true,
                iconSize: 50,
              ),
              // SignInButton(
              //   button: Button.AppleDark,
              //   small: true,
              // ),
              SizedBox(height: 10),
              SignInButton(
                button: Button.TikTok,
                small: true,
              ),
              SizedBox(height: 10),
              SignInButton(button: Button.Email),
              SizedBox(height: 10),
              CustomSignInButton(
                height: 80,
                width: 80,
                // text: ' ',
                iconLeftPadding: 2,
                iconTopPadding: 2,
                iconSize: 62,
                customIcon: FontAwesomeIcons.facebookF,
                iconColor: Colors.white,
                buttonColor: Colors.blue.shade700,
              )
            ],
          ),
        ),
      ),
    );
  }
}


lib/button_builder.dart
// ignore_for_file: prefer_const_constructors_in_immutables, prefer_const_constructors
import 'package:flutter/material.dart';

class CustomSignInButton extends StatelessWidget {
  final double borderRadius;
  final Color buttonColor;
  final IconData customIcon;
  final double elevation;
  final double height;
  final Color iconColor;
  final double iconSize;
  final Color imageBackgroundColor;
  final EdgeInsetsGeometry imageMargin;
  final EdgeInsetsGeometry imagePadding;
  final double iconLeftPadding;
  final double iconTopPadding;
  final double imageSize;
  final bool mini;
  final bool small;
  final Function()? onPressed;
  final AssetImage setAssetImage;
  final Color splashColor;
  final String text;
  final Color textColor;
  final double textSize;
  final bool useGoogleImage;
  final bool useGradient;
  final bool showText;
  final Gradient setGradient;
  final double width;

  const CustomSignInButton({
    this.borderRadius = 5,
    this.buttonColor = const Color(0xFFEEEEEE),
    this.customIcon = Icons.new_releases_rounded,
    this.elevation = 0,
    this.height = 40,
    this.iconColor = Colors.grey,
    this.iconSize = 24,
    this.imageBackgroundColor = Colors.transparent,
    this.imageMargin = const EdgeInsets.all(4),
    this.imagePadding = const EdgeInsets.all(4),
    this.iconLeftPadding = 0,
    this.iconTopPadding = 0,
    this.imageSize = 25,
    Key? key,
    this.mini = false,
    this.small = false,
    this.onPressed,
    this.setAssetImage = const AssetImage(''),
    this.splashColor = Colors.white,
    this.text = 'New Button',
    this.textColor = Colors.black87,
    this.textSize = 15,
    this.useGoogleImage = false,
    this.useGradient = false,
    this.showText = true,
    this.setGradient = const LinearGradient(colors: [Colors.grey, Colors.grey]),
    this.width = double.infinity,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Flexible(
      fit: FlexFit.loose,
      child: Container(
        height: mini
            ? 40
            : small
                ? 55
                : height,
        width: mini
            ? 40
            : small
                ? 55
                : width,
        decoration: !useGradient
            ? BoxDecoration(
                color: buttonColor,
                borderRadius: BorderRadius.circular(borderRadius),
              )
            : BoxDecoration(
                gradient: setGradient,
                borderRadius: BorderRadius.circular(borderRadius)),
        child: MaterialButton(
          minWidth: 40,
          padding: EdgeInsets.only(
            left: small ? -2 : iconLeftPadding,
            top: small ? 4 : iconTopPadding,
            bottom: small ? 4 : 0,
          ),
          splashColor: splashColor,
          shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(borderRadius)),
          child: Row(
            mainAxisAlignment: MainAxisAlignment.center,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Container(
                  decoration: BoxDecoration(
                    color: imageBackgroundColor,
                    borderRadius: BorderRadius.circular(borderRadius),
                  ),
                  padding: imagePadding,
                  margin: imageMargin,
                  child: Container(
                    alignment: Alignment.topLeft,
                    child: useGoogleImage
                        ? Image(
                            image: AssetImage('assets/google.png',
                                package: 'custom_signin_buttons'),
                            height: small ? 34 : imageSize,
                          )
                        : setAssetImage.assetName.isNotEmpty
                            ? Image(
                                image: setAssetImage,
                                height: small ? 34 : imageSize,
                              )
                            : Icon(
                                customIcon,
                                color: iconColor,
                                size: small ? 32 : iconSize,
                              ),
                  )),
              (mini || small)
                  ? SizedBox()
                  : Flexible(
                      child: Center(
                        child: Text(showText ? text : '',
                            style: TextStyle(
                              fontSize: textSize,
                              fontWeight: FontWeight.w600,
                              color: textColor,
                            )),
                      ),
                    ),
            ],
          ),
          onPressed: onPressed ?? () {},
        ),
      ),
    );
  }
}


lib/button_data.dart
// ignore_for_file: prefer_const_constructors_in_immutables, prefer_const_constructors

import 'package:flutter/material.dart';
import 'package:custom_signin_buttons/custom_signin_buttons.dart';

class SignInButton extends StatelessWidget {
  final Button button;
  final double borderRadius;
  final double height;
  final double iconSize;
  final bool mini;
  final bool small;
  final bool showText;
  final Function()? onPressed;
  final Color splashColor;
  final String? text;
  final double textSize;
  final double width;
  final bool spanish;

  SignInButton({
    Key? key,
    this.button = Button.YoutubeDark,
    this.borderRadius = 5,
    this.height = 40,
    this.iconSize = 24,
    this.mini = false,
    this.small = false,
    this.showText = true,
    this.onPressed,
    this.splashColor = Colors.white30,
    this.text = '',
    this.textSize = 15,
    this.width = double.infinity,
    this.spanish = false,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    var texto = '';
    spanish ? texto = 'Iniciar Sesion con' : texto = 'Sign in with';
    switch (button) {
      case Button.Apple:
      case Button.AppleDark:
        return CustomSignInButton(
            key: ValueKey('Apple'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Apple',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor:
                button == Button.Apple ? Color(0xFFE9E9E9) : Color(0xFF222222),
            textColor:
                button == Button.Apple ? Colors.black87 : Color(0xFFF3F2F1),
            customIcon: FontAwesomeIcons.apple,
            iconColor:
                button == Button.Apple ? Color(0xFF050505) : Color(0xFFF3F2F1),
            iconLeftPadding: mini ? 0 : 5,
            imagePadding: EdgeInsets.symmetric(vertical: 3),
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Discord:
        return CustomSignInButton(
            key: ValueKey('Discord'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Discord',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFF5663F7),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.discord,
            iconLeftPadding: mini ? 0 : 9,
            imagePadding: EdgeInsets.symmetric(vertical: 3),
            imageMargin: EdgeInsets.only(right: 5, top: 4),
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Email:
        return CustomSignInButton(
            key: ValueKey('Email'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Email',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Colors.black12,
            textColor: Colors.black,
            customIcon: FontAwesomeIcons.envelope,
            iconColor: Colors.black87,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Facebook:
      case Button.FacebookNew:
        return CustomSignInButton(
            key: ValueKey('Facebook'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Facebook',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: button == Button.FacebookNew
                ? Color(0xFF1877f2)
                : Color(0xFF3B5998),
            textColor: Colors.white,
            customIcon: button == Button.FacebookNew
                ? FontAwesomeIcons.facebook
                : FontAwesomeIcons.facebookF,
            // iconSize: button == Button.FacebookNew ? 28 : 25,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.GitHub:
        return CustomSignInButton(
            key: ValueKey('GitHub'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto GitHub',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFF444444),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.github,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Google:
      case Button.GoogleBlack:
      case Button.GoogleBlue:
        return CustomSignInButton(
            key: ValueKey('Google'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Google',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: button == Button.GoogleBlue
                ? Color(0xFF397AF3)
                : button == Button.GoogleBlack
                    ? Color(0xFF222222)
                    : Color(0xFFF6F6F6),
            textColor:
                (button == Button.GoogleBlue || button == Button.GoogleBlack)
                    ? Colors.white
                    : Colors.black87,
            imageBackgroundColor:
                (button == Button.GoogleBlue || button == Button.GoogleBlack)
                    ? Colors.white
                    : Colors.transparent,
            useGoogleImage: true,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.GooglePlay:
        return CustomSignInButton(
            key: ValueKey('GooglePlay'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto PlayStore',
            showText: showText,
            height: height,
            width: width,
            setAssetImage: AssetImage(
              'assets/playstore.png',
              package: 'custom_signin_buttons',
            ),
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFFF6F6F6),
            textColor: Colors.black87,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Instagram:
      case Button.InstagramGradient:
        return CustomSignInButton(
            key: ValueKey('Instagram'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Instagram',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFFDD2A7B),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.instagram,
            useGradient: button == Button.InstagramGradient ? true : false,
            setGradient: LinearGradient(
                begin: Alignment.bottomLeft,
                end: Alignment.topRight,
                colors: const [
                  Color(0xFFF58529),
                  Color(0xFFDD2A7B),
                  Color(0xFF8134AF),
                  Color(0xFF515BD4)
                ],
                stops: const [
                  0,
                  0.40,
                  0.70,
                  1
                ]),
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.LinkedIn:
        return CustomSignInButton(
            key: ValueKey('LinkedIn'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto LinkedIn',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFF007BB5),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.linkedinIn,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Pinterest:
        return CustomSignInButton(
            key: ValueKey('Pinterest'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Pinterest',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFFCB1F27),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.pinterest,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.PlayStation:
      case Button.PlayStationGradient:
        return CustomSignInButton(
            key: ValueKey('PlayStation'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto PSN',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFF00439c),
            useGradient: button == Button.PlayStationGradient ? true : false,
            setGradient: LinearGradient(
                begin: Alignment.bottomCenter,
                end: Alignment.topCenter,
                colors: const [
                  Color(0xFF00246E),
                  Color(0xFF0172D0),
                ],
                stops: const [
                  0.1,
                  0.9
                ]),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.playstation,
            imagePadding: EdgeInsets.symmetric(vertical: 3),
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Reddit:
      case Button.RedditGradient:
        return CustomSignInButton(
            key: ValueKey('Reddit'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Reddit',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFFFF4500),
            useGradient: button == Button.RedditGradient ? true : false,
            setGradient: LinearGradient(
                begin: Alignment.bottomCenter,
                end: Alignment.topCenter,
                colors: const [
                  Color(0xFFef131a),
                  Color(0xFFFF4500),
                ],
                stops: const [
                  0.1,
                  0.4
                ]),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.reddit,
            imagePadding: EdgeInsets.symmetric(vertical: 3),
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Spotify:
      case Button.SpotifyLight:
        return CustomSignInButton(
            key: ValueKey('Spotify'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Spotify',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: button == Button.SpotifyLight
                ? Color(0xFF1DB954)
                : Color(0xFF222222),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.spotify,
            iconSize: iconSize,
            iconColor: button == Button.SpotifyLight
                ? Colors.white
                : Color(0xFF1DB954),
            textSize: textSize,
            onPressed: onPressed);
      case Button.Steam:
      case Button.SteamGradient:
        return CustomSignInButton(
            key: ValueKey('Steam'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Steam',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            useGradient: button == Button.SteamGradient ? true : false,
            setGradient: LinearGradient(
              begin: Alignment.topCenter,
              end: Alignment.bottomCenter,
              colors: const [
                Color(0xFF0a1a35),
                Color(0xFF133669),
                Color(0xFF1384b5)
              ],
              stops: const [0, 0.65, 1],
            ),
            buttonColor: Color(0xFF133669),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.steamSymbol,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.TikTok:
        return CustomSignInButton(
            key: ValueKey('TikTok'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto TikTok',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            setAssetImage: AssetImage('assets/tiktok.png',
                package: 'custom_signin_buttons'),
            imageSize: 24,
            buttonColor: Colors.black,
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.twitter,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Twitch:
      case Button.TwitchLight:
        return CustomSignInButton(
            key: ValueKey('Twitch'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Twitch',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: button == Button.TwitchLight
                ? Color(0xFF9146FF)
                : Color(0xFFFBFBFB),
            textColor:
                button == Button.TwitchLight ? Colors.white : Colors.black,
            customIcon: FontAwesomeIcons.twitch,
            iconColor:
                button == Button.TwitchLight ? Colors.white : Color(0xFF9146FF),
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Twitter:
        return CustomSignInButton(
            key: ValueKey('Twitter'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Twitter',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: Color(0xFF1DA1F2),
            textColor: Colors.white,
            customIcon: FontAwesomeIcons.twitter,
            iconColor: Colors.white,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Xbox:
      case Button.XboxDark:
        return CustomSignInButton(
            key: ValueKey('Xbox'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Xbox',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor:
                button == Button.Xbox ? Color(0xFFFBFBFB) : Color(0xFF107C10),
            textColor:
                button == Button.Xbox ? Colors.black87 : Color(0xFFF3F2F1),
            customIcon: FontAwesomeIcons.xbox,
            iconColor:
                button == Button.Xbox ? Color(0xFF107C10) : Color(0xFFF3F2F1),
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
      case Button.Youtube:
      case Button.YoutubeDark:
        return CustomSignInButton(
            key: ValueKey('Youtube'),
            mini: mini,
            small: small,
            text: text!.isNotEmpty ? text! : '$texto Youtube',
            showText: showText,
            height: height,
            width: width,
            splashColor: splashColor,
            borderRadius: borderRadius,
            buttonColor: button == Button.Youtube
                ? Color(0xFFFBFBFB)
                : Color(0xFFFF0000),
            textColor: button == Button.Youtube ? Colors.black87 : Colors.white,
            iconColor: button == Button.Youtube
                ? Color(0xFFFF0000)
                : Color(0xFFFBFBFB),
            customIcon: FontAwesomeIcons.youtube,
            iconSize: iconSize,
            textSize: textSize,
            onPressed: onPressed);
    }
  }
}
